### PR TITLE
babel 6

### LINKED ISSUE #8: Babel 6 support
https://babeljs.io/blog/2015/10/29/6.0.0/

### PR DESCRIPTION
**Pull request by TaopaiC**
I tried to upgrade with babel 6.

I also tested this plugin in a small plugin
But since all the generated source changed, I modified all the expected fixture files. Please help review the generated sources.

Fixes #8

### CODE REVIEW

**Comment by megawac** [comment]
Awesome! Thanks for looking into this I'll review as soon as I can

File: package.json [hunk 1]
```diff
@@ -24,6 +28,13 @@
...
   },
   "babel": {
-    "ignore": []
+    "ignore": [],
+    "presets": [
+      "es2015",
+      "es2015-node4"
+    ],
+    "plugins": [
+      "transform-es2015-modules-commonjs"
```
Comment (megawac) [inline package.json]: is it preferred to do this in the package.json or .babelrc?

File: test/fixtures/lodash-fp-mix/expected.js [hunk 2]
```diff
@@ -1,34 +1,34 @@
...
-var _lodashFp = (0, _lodashFpConvert3['default'])({
-  'map': _lodashCollectionMap3['default'],
-  'add': _lodashMathAdd3['default'],
-  'take': _lodashArrayTake3['default'],
-  'reject': _lodashCollectionReject3['default']
-});
+function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }
 
-function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { 'default': obj }; }
+const _lodashFp = (0, _convert2.default)({
```
Comment (megawac) [inline test/fixtures/lodash-fp-mix/expected.js]: why did it leave const in?

File: package.json (hunk 1 above)
Comment (TaopaiC) [inline package.json]: I have no idea which it better. Do you prefer to do this in .babelrc?

File: test/fixtures/lodash-fp-mix/expected.js (hunk 2 above)
Comment (TaopaiC) [inline test/fixtures/lodash-fp-mix/expected.js]: Not sure why _lodashFp is not processed, I change the variableDeclaration kind to 'var' now.

**Comment by TaopaiC** [comment]
`npm run push` won't work since babel-plugin was removed. Is it ok to push in another way?

**Comment by megawac** [comment]
Yeah, iirc plugin just builds the source files with babel

**Comment by jdalton** [comment]
@TaopaiC What's es2015-node4 provide that the es2015 preset doesn't?

**Comment by megawac** [comment]
It adds some cjs presets. Not sure if its necessary tests pass without it

**Comment by jdalton** [comment]
It looks like the [es2015 preset](http://babeljs.io/docs/plugins/preset-es2015/) includes all the transforms of [babel-preset-es2015-node4](https://github.com/jbach/babel-preset-es2015-node4/blob/master/index.js).

**Comment by TaopaiC** [comment]
@jdalton you are correct. babel-preset-es2015 is enough.
I added babel-preset-es2015-node4 first, and then added babel-preset-es2015 after test failed on node v0.12.

**Comment by jdalton** [comment]
Hey @TaopaiC! I know this is a bit after the fact but could you sign the [jQuery Foundation CLA](https://contribute.jquery.org/CLA/) if you haven't already.

**Comment by TaopaiC** [comment]
just signed. Thank you

**Pull request merged**